ffi/cpp/include/locale.hpp

```diff
@@ -0,0 +1,39 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#ifndef ICU4X_LOCALE_HPP
+#define ICU4X_LOCALE_HPP
+
+#include <algorithm>
```

Comment:
question: Is `<algorithm>` needed here? I think that `std::optional` is defined in `<optional>`

Reply:
It's for `std::swap`

Reply:
I would prefer using `#include <utility>` here if that's the only dependency on `<algorithm>` since `<utility>` defines `std::swap` and is more lightweight.